src/webgpu/api/validation/createBindGroup.spec.ts

```diff
@@ -204,49 +217,55 @@ g.test('texture_must_have_correct_component_type')
     }
   });
 
-// TODO: Write test for all dimensions.
-g.test('texture_must_have_correct_dimension').fn(async t => {
-  const bindGroupLayout = t.device.createBindGroupLayout({
-    entries: [
-      {
-        binding: 0,
-        visibility: GPUShaderStage.FRAGMENT,
-        texture: {},
-      },
-    ],
-  });
+g.test('texture_must_have_correct_dimension')
+  .desc(
+    `
+    Test that bound texture views match the dimensions supplied in the BindGroupLayout
+    - Test for every GPUTextureViewDimension`
+  )
+  .cases(poptions('viewDimension', kTextureViewDimensions))
+  .fn(async t => {
+    const { viewDimension } = t.params;
+    const bindGroupLayout = t.device.createBindGroupLayout({
+      entries: [
+        {
+          binding: 0,
+          visibility: GPUShaderStage.FRAGMENT,
+          texture: { viewDimension },
+        },
+      ],
+    });
 
-  const goodDescriptor = {
-    size: { width: 16, height: 16, depthOrArrayLayers: 1 },
-    format: 'rgba8unorm' as const,
-    usage: GPUTextureUsage.SAMPLED,
-  };
+    const textureDescriptor = {
+      size: { width: 16, height: 16, depthOrArrayLayers: 6 },
+      format: 'rgba8unorm' as const,
+      usage: GPUTextureUsage.SAMPLED,
+    };
 
-  // Control case
-  t.device.createBindGroup({
-    entries: [{ binding: 0, resource: t.device.createTexture(goodDescriptor).createView() }],
-    layout: bindGroupLayout,
-  });
+    const texture = t.device.createTexture(textureDescriptor);
 
-  // Mismatched texture binding formats are not valid.
-  const badDescriptor = clone(goodDescriptor);
-  badDescriptor.size.depthOrArrayLayers = 2;
+    for (const dimension of kTextureViewDimensions) {
+      const shouldError = viewDimension !== dimension;
+      const arrayLayerCount = dimension === '2d' ? 1 : undefined;
+      const viewDescriptor = { dimension, arrayLayerCount };
 
-  t.expectValidationError(() => {
-    t.device.createBindGroup({
-      entries: [{ binding: 0, resource: t.device.createTexture(badDescriptor).createView() }],
-      layout: bindGroupLayout,
-    });
+      t.expectValidationError(() => {
+        t.device.createBindGroup({
+          entries: [{ binding: 0, resource: texture.createView(viewDescriptor) }],
```

Comment:
This expectValidationError currently captures multiple calls (createBindGroup and createView). Need to move the createView outside so we know we are testing the right thing. (Also I think createView might actually fail in some cases, not positive)

Reply:
Moved `createView()` outside the validation error check, but now every case of the test fails because they all attempt to create every view variant to test compatibility and Dawn doesn't support 1D or 3D views yet. That's a little discouraging, but I'm not sure what a better pattern would be. I could check and see if `createView()` fails and skip that iteration of loop, since this isn't intended to test `createView()`? I'd appreciate your thoughts.

Reply:
Ah, true. How about replacing the loop with subcases? The setup part of the test will end up running more, but that seems fine.

Reply:
Oh, sorry I missed this. Github didn't include it in the page auto-refresh for some reason. I can do that.

Reply:
And done.